tools/src/launcher/Main.hs

```diff
@@ -100,6 +100,20 @@ type M a = (HasConfigurations, HasCompileInfo) => Log.LoggerNameBox IO a
 
 data Executable = EWallet | ENode | EUpdater
 
+-- Node, its args, node log
+data NodeData = NodeData
+    { _nodePath    :: FilePath
+    , _nodeArgs    :: [Text]
+    , _nodeLogPath :: Maybe FilePath }
```

Comment:
Why did you use `_` prefix? The only reason to use `_` prefix I am aware of is when you also use `makeLenses` (because it strips `_`). But you don't use `_` for this type.
Our common convention is to use abbreviation of the type as prefix, i. e. `nd` in this case, so the fields would be called: `ndPath`, `ndArgs` and `ndLogPath`.
Also our style guides suggest making all fields strict by default (using `!`), i. e. when there are no reasons to make them lazy.

Reply:
Ah, probably that's because you don't use these names and it leads to warnings.
Well, to be honest, I think it's better to use these names. Explicitly assigning values to different fields makes code clearer, in my opinion.

Reply:
> Ah, probably that's because you don't use these names and it leads to warnings.

Right. I used proper assigning instead of pattern matching so I don't get warnings no more.